packages/expo-updates/android/src/main/java/expo/modules/updates/loader/FileDownloader.kt

```diff
@@ -110,6 +174,188 @@ class FileDownloader(
     }
   }
 
+  private fun isPatchResponse(responseBody: ResponseBody): Boolean {
+    val mediaType: MediaType? = responseBody.contentType()
+    return mediaType?.let { it.type == "application" && it.subtype == "vnd.bsdiff" } ?: false
+  }
+
+  @Throws(IOException::class)
+  private fun applyPatchAndVerify(
+    asset: AssetEntity,
+    request: Request,
+    responseBody: ResponseBody,
+    destination: File,
+    updatesDirectory: File,
+    expectedBase64URLEncodedSHA256Hash: String?
+  ): ByteArray {
+    val launchAssetContext = prepareAssetForDiff(
+      asset,
+      request,
+      responseBody,
+      updatesDirectory
+    )
+
+    val requestedUpdateId = request.header(EXPO_REQUESTED_UPDATE_ID_HEADER)
```

Comment:
slight preference to pass these in as separate args rather than re-extract them from the request headers. that way it's clearer that they are required.